pkg/querier/frontend/frontend.go

```diff
@@ -179,6 +179,10 @@ func (f *Frontend) Handler() http.Handler {
 }
 
 func (f *Frontend) handle(w http.ResponseWriter, r *http.Request) {
+	// create buffer for later use in case of slow query
+	// to obtain request's form values
+	var buf bytes.Buffer
+	r.Body = ioutil.NopCloser(io.TeeReader(r.Body, &buf))
```

Comment:
Why `NopCloser` here?

Reply:
Here we never close `r.Body`, so perhaps we should do that in `defer` statement at least.

Reply:
Good catch, added defer for `r.Body`